plugins/catalog-react/src/hooks/useEntityPermission.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { Permission } from '@backstage/plugin-permission-common';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { useEntity } from './useEntity';
+
+/**
+ * A thin wrapper around the
+ * {@link @backstage/plugin-permission-react#usePermission} hook which uses the
+ * current entity in context to make an authorization request for the given
+ * permission.
+ * @public
+ */
+export function useEntityPermission(permission: Permission) {
+  const { entity } = useEntity();
+  if (!entity) {
+    throw new Error('No entity in current context.');
+  }
+  return usePermission(permission, stringifyEntityRef(entity));
```

Comment:
Requiring the full entity as input here will block the permission request until the entity is loaded. This seems fine for the "unregister component" UX because it's hidden behind a menu — just calling out that other views may want to optimize the request flow by loading the entity + permissions in parallel. Could add a comment about using `usePermission` directly in those cases.

Reply:
Done.